test/unit/size_check.c

```diff
@@ -0,0 +1,59 @@
+#include "test/jemalloc_test.h"
+
+#include "jemalloc/internal/safety_check.h"
+
+bool fake_abort_called;
+void fake_abort(const char *message) {
+	(void)message;
+	fake_abort_called = true;
+}
+
+TEST_BEGIN(test_invalid_size_sdallocx) {
+	test_skip_if(!config_opt_size_checks);
+	safety_check_set_abort(&fake_abort);
+
+	fake_abort_called = false;
+	void *ptr = malloc(SC_SMALL_MAXCLASS);
+	assert_ptr_not_null(ptr, "Unexpected failure");
+	sdallocx(ptr, SC_LARGE_MINCLASS, 0);
```

Comment:
Well, to be super picky here, I think checking for a different small size rather than a large size might be slightly better. One reason is that in most of the real cases, there will be a mismatch within the small size classes. Another reason is that these tests can still pass if there isn't the `alloc_ctx->szind != dbg_ctx.szind` branch at all in the `maybe_check_alloc_ctx()` function, which is the more important branch, and checking for small size mismatch can improve the test coverage.

Reply:
Will do.